convert2rhel/actions/rhel_compatible_kernel.py

```diff
@@ -0,0 +1,141 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+
+from convert2rhel import actions
+from convert2rhel.pkghandler import get_installed_pkg_objects, get_pkg_fingerprint
+from convert2rhel.systeminfo import system_info
+from convert2rhel.utils import run_subprocess
+
+
+logger = logging.getLogger(__name__)
+
+# The kernel version stays the same throughout a RHEL major version
+COMPATIBLE_KERNELS_VERS = {
+    7: "3.10.0",
+    8: "4.18.0",
+}
+BAD_KERNEL_RELEASE_SUBSTRINGS = ("uek", "rt", "linode")
+
+
+class RhelCompatibleKernel(actions.Action):
+    id = "RHEL_COMPATIBLE_KERNEL"
+
+    def run(self):
+        """Ensure the booted kernel is signed, is standard (not UEK, realtime, ...), and has the same version as in RHEL.
+        By requesting that, we can be confident that the RHEL kernel will provide the same capabilities as on the
+        original system.
+        """
+        logger.task("Prepare: Check kernel compatibility with RHEL")
```

Comment:
We just recently realized that it is a good idea to do this:

```suggestion
        super(RhelCompatibleKernel, self).run()
        logger.task("Prepare: Check kernel compatibility with RHEL")
```

without calling super, the default successful return values are not added to the class attributes.

Reply:
Placing the decorator on top of `run()` seems elegant, too. 

```
@actions._action_defaults_to_success
def run(self): 
```

But either way, both will work